agent-core/src/main/java/com/newrelic/agent/android/stats/TicToc.java

```diff
@@ -32,7 +33,31 @@ public long toc() {
     }
 
     public long peek() {
-        return (state == State.STARTED) ? System.currentTimeMillis() - startTime : 0;
+        return (state == State.STARTED) ? System.nanoTime() - startTime : 0;
+    }
+
+    public long getStartTime() {
```

Comment:
Do we need accessors to set start/end time? The class encapsulates a time span, so I wouldn't think either value would be manually changed by the caller.

Reply:
I mainly use the getter/setter to verify unit tests.

Reply:
Give the fields and methods package private or protected access, and tests can access them.